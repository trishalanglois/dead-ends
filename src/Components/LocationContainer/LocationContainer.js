import React from 'react';
import './LocationContainer.scss';

const LocationContainer = () => {
  return (
    <h2> This is my location container ay ay ayyyyyy </h2>
  )
}

export default LocationContainer
