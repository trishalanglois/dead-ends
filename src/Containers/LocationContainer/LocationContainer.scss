@import '../../scss/variables';


.locations-container-subheader {
  display: flex;
  justify-content: space-between;
  background-color: $dark-color;
  height: 100%;
}


.location-subheader {
  font-family: $accent-font;
  font-size: 3em;
  margin: 0px 2em 1em 2em;
  color: $light-grey;
  font-weight: 200;
  width: auto;
}

.favorites-button {
  @extend .location-subheader;
  font-size: 2em;
}

.main-section {
  background-image: url('../../images/background.jpg');
  background-size: cover;
  background-position: center;
  height: 88vh;
  width: 100vw;
}

.locations-cards-container {
  overflow: auto;
  width: 50%;
  height: 100%;
}
