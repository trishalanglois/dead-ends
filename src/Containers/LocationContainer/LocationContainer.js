import React from 'react';
import './LocationContainer.scss';
import LocationCard from '../LocationCard/LocationCard';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export const LocationContainer = ({ locations, location, mapLocation }) => {
  const showLocations = locations.map((spookyLocation, i) => {
    return (
      <>
        <LocationCard
          key={Math.random() * i}
          cardLocation={spookyLocation}
        />
      </>
    )
  })

  return (
    <main className="locations-container">
      <div className="locations-container-subheader">
        <h2 className="location-subheader">{location}</h2>
        <Link to={'/favorites'}>
          <h3 className="favorites-button">View Favorites</h3>
        </Link>
      </div>
      <div className="main-section">
          {showLocations}
      </div>
    </main>
  )
}

export const mapStateToProps = state => ({
  location: state.location,
  locations: state.spookyLocations,
  mapLocation: state.currentMap
})

export default connect(mapStateToProps)(LocationContainer);

LocationContainer.propTypes = {
  location: PropTypes.string,
  locations: PropTypes.array,
  mapLocation: PropTypes.object
}
