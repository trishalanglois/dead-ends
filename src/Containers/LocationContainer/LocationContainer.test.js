import React from 'react';
import { LocationContainer, mapStateToProps } from './LocationContainer';
import { shallow } from 'enzyme';

describe('LocationContainer', () => {
  let wrapper;
  it('should match snapshot', () => {
    const mockProps = {
    location: 'Denver, CO',
    locations: [{
      "city": "Denver",
      "country": "United States",
      "description": "BOO!",
      "location": "Ada Cemetery",
      "state": "Colorado",
      "state_abbrev": "CO",
      "longitude": -85.50489309999999,
      "latitude": 42.9621061,
      "city_longitude": -85.4954803,
      "city_latitude": 42.960727
    },
    {
      "city": "Denver",
      "country": "United States",
      "description": "Spooky stuff.",
      "location": "North Adams Rd.",
      "state": "Colorado",
      "state_abbrev": "CO",
      "longitude": -84.3818434,
      "latitude": 41.9714248,
      "city_longitude": -84.3471684,
      "city_latitude": 41.986434
    }],
    mapLocation: {
      "city": "Denver",
      "country": "United States",
      "description": "BOO!",
      "location": "Ada Cemetery",
      "state": "Colorado",
      "state_abbrev": "CO",
      "longitude": -85.50489309999999,
      "latitude": 42.9621061,
      "city_longitude": -85.4954803,
      "city_latitude": 42.960727
    }
  };
    wrapper = shallow(<LocationContainer
    locations={mockProps.locations}
    location={mockProps.location}
    mapLocation={mockProps.mapLocation}
    />);

    expect(wrapper).toMatchSnapshot();
  })

  it('should return an object with the location, locations, and mapLocations data', () => {
    const mockState = {
      location: "Denver, CO",
      spookyLocations: [{
        "city": "Denver",
        "country": "United States",
        "description": "BOO!",
        "location": "Ada Cemetery",
        "state": "Colorado",
        "state_abbrev": "CO",
        "longitude": -85.50489309999999,
        "latitude": 42.9621061,
        "city_longitude": -85.4954803,
        "city_latitude": 42.960727
      },
      {
        "city": "Denver",
        "country": "United States",
        "description": "Spooky stuff.",
        "location": "North Adams Rd.",
        "state": "Colorado",
        "state_abbrev": "CO",
        "longitude": -84.3818434,
        "latitude": 41.9714248,
        "city_longitude": -84.3471684,
        "city_latitude": 41.986434
      }],
      currentMap: {
        "city": "Denver",
        "country": "United States",
        "description": "BOO!",
        "location": "Ada Cemetery",
        "state": "Colorado",
        "state_abbrev": "CO",
        "longitude": -85.50489309999999,
        "latitude": 42.9621061,
        "city_longitude": -85.4954803,
        "city_latitude": 42.960727
      }
    };
    const expected = {
      location: "Denver, CO",
      locations: [{
        "city": "Denver",
        "country": "United States",
        "description": "BOO!",
        "location": "Ada Cemetery",
        "state": "Colorado",
        "state_abbrev": "CO",
        "longitude": -85.50489309999999,
        "latitude": 42.9621061,
        "city_longitude": -85.4954803,
        "city_latitude": 42.960727
      },
      {
        "city": "Denver",
        "country": "United States",
        "description": "Spooky stuff.",
        "location": "North Adams Rd.",
        "state": "Colorado",
        "state_abbrev": "CO",
        "longitude": -84.3818434,
        "latitude": 41.9714248,
        "city_longitude": -84.3471684,
        "city_latitude": 41.986434
      }],
      mapLocation: {
        "city": "Denver",
        "country": "United States",
        "description": "BOO!",
        "location": "Ada Cemetery",
        "state": "Colorado",
        "state_abbrev": "CO",
        "longitude": -85.50489309999999,
        "latitude": 42.9621061,
        "city_longitude": -85.4954803,
        "city_latitude": 42.960727
      }
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected)
  })
})
