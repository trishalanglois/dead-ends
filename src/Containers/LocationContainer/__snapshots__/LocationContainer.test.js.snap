// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationContainer should match snapshot 1`] = `
<main
  className="locations-container"
>
  <div
    className="locations-container-subheader"
  >
    <h2
      className="location-subheader"
    >
      Denver, CO
    </h2>
    <Link
      to="/favorites"
    >
      <h3
        className="favorites-button"
      >
        View Favorites
      </h3>
    </Link>
  </div>
  <div
    className="main-section"
  >
    <Connect(LocationCard)
      cardLocation={
        Object {
          "city": "Denver",
          "city_latitude": 42.960727,
          "city_longitude": -85.4954803,
          "country": "United States",
          "description": "BOO!",
          "latitude": 42.9621061,
          "location": "Ada Cemetery",
          "longitude": -85.50489309999999,
          "state": "Colorado",
          "state_abbrev": "CO",
        }
      }
      key="0"
    />
    <Connect(LocationCard)
      cardLocation={
        Object {
          "city": "Denver",
          "city_latitude": 41.986434,
          "city_longitude": -84.3471684,
          "country": "United States",
          "description": "Spooky stuff.",
          "latitude": 41.9714248,
          "location": "North Adams Rd.",
          "longitude": -84.3818434,
          "state": "Colorado",
          "state_abbrev": "CO",
        }
      }
      key="0.9178474132219165"
    />
  </div>
</main>
`;
